### PR TITLE
add contenttype id to list picker #894

### LINKED ISSUE #894: add contenttype id to list picker
**Please check out the [documentation](https://pnp.github.io/sp-dev-fx-controls-react/) to see if your question is already addressed there. This will help us ensure our documentation is up to date.**

#### Category
[ x] Enhancement

[ ] Bug

[ ] Question

#### Version

Please specify what version of the library you are using: [latest        ]
Could we add a contentTypeID to the listpicker control. The ide would be that the list picker would only let you select lists that contained the selected content type.

### PR DESCRIPTION
**Pull request by russgove**
| Q               | A
| --------------- | ---
| Bug fix?        | [ ]
| New feature?    | [x ]
| New sample?      | [ ]
| Related issues?  | fixes #894

#### What's in this Pull Request?

added contentTypeId to listpicker

### CODE REVIEW

**Comment by joelfmrodrigues** [comment]
@russgove this build error is not specific to your code, but to a specific version of an npm module that is causing issues: I was able to reproduce this on my machine. Give me just a moment while I try to figure out the best way to address it as there is already a PR where this was addressed

**Comment by joelfmrodrigues** [comment]
@russgove if you now get the latest changes from dev, it should fix the npm install error as the package-lock.json file was updated. No need to make changes to the package-lock.json file within your PR

**Comment by russgove** [comment]
Thanks joel, i have other stuff im working on on my local machine. i did
not make a separate branch for it(ouch!). Is  ther a way i can do this from
the github portal?

**Comment by joelfmrodrigues** [comment]
Hi @russgove sorry for the delay...a lot of work going on at my side at the moment.
I will try to have a look tomorrow, I'm sure we will find a simple way to resolve this.

**Comment by russgove** [comment]
hey , i just included the latest packkage lock and that resolved it.

**Comment by russgove** [comment]
added another commit to my PR to add a selectedSites property to the SItePicker.

File: src/controls/listPicker/ListPicker.tsx [hunk 1]
```diff
@@ -1,13 +1,13 @@
-import * as React from 'react';
```
Comment (joelfmrodrigues) [inline src/controls/listPicker/ListPicker.tsx:1]: @russgove for future PRs, if possible please avoid re-ording imports. I assume this was done by a VS code extension?

**Comment by joelfmrodrigues** [comment]
@russgove sorry for the delay reviewing this, I was extremely busy at work over the last 2 weeks.
While checking the code, I noticed that there are a lot of formatting changes (spaces) and imports re-ordering that seem to have been done by VS code, possibly an extension. This makes the review more complex and in theory, PRs should not include formatting changes to previous code.
There is also the case of the site picker changes being mixed (I know it was an accident)
Any chance you could create two new separate PRs? One for the list picker and one for the site picker? If you don't have the chance to do this, I can try to help

**Comment by russgove** [comment]
Hi Joel,
How do i do that? i made a pr for one change, then did a git push from my local repo to my github and it automatically adds what i just pushed to my existing pr.

**Comment by joelfmrodrigues** [comment]
Hi @russgove ,

I think there are multiple ways to address this, but personally, I think that undoing the commits may be the easiest approach. Then stash them and create a new clean branch from that one that you can use for a new PR. You can then use the stashed changes on a new branch (from dev) and also create a new PR.

A different alternative, considering that there are only a few files changed, is to just create two new branches from dev and manually copy the changes from this PR into each branch as appropriate and submit 2 new PRs

**Comment by joelfmrodrigues** [comment]
@russgove not sure if I got your question, but only commits to the branch associated with the PR get merged as part of that PR. In this case, for instance, only commits to your dev branch.
Personally, I always use a new branch for each PR and I delete it after the PR is merged. And I keep my dev branch in sync with the dev branch of the main repository (in this case the PnP controls repo). By doing this, I can just switch to dev, click the VS code button to sync, and I get all the commits from everyone else that were merged.

Please let me know if you need help splitting the PRs. More than happy to help if needed

**Comment by russgove** [comment]
I could use some help splitting the PRs. My git knowledge is limited.

I’m in New york time zone, where are you?
Russell

**Comment by AJIXuMuK** [comment]
@russgove @joelfmrodrigues  - I've merged it manually

**Pull request merged**